src/main/java/Duke/tasks/Deadline.java

```diff
@@ -0,0 +1,51 @@
+package duke.tasks;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+
+import duke.ui.DukeException;
+
+/**
+ * Represents the deadline of task.
+ */
+public class Deadline extends Task {
+    protected LocalDate date;
+    protected LocalTime time;
+
+    /**
+     * Constructor for Deadline.
+     *
+     * @param description description of the task.
+     * @param by by when the deadline end.
+     */
+    public Deadline(String description, String by) throws DukeException {
+        super(description);
+        this.date = super.getTaskDate(by);
+        this.time = super.getTaskTime(by);
+    }
+
+    /**
+     * Strings representation of Deadline in the save file.
```

Comment:
I feel that `Strings` is not widely used as a verb. A better way to describe this will be `Returns the string representation of ...`. I've noted the same issue in other places too.